history.go

```diff
@@ -52,25 +57,33 @@ func Run() error {
 		if err != nil {
 			return fmt.Errorf("error reading the input")
 		}
-		//fmt.Println(text)
 		if text == "exit" || text == "quit" {
 			break
 		}
+		cmdHistory = append(cmdHistory, fmt.Sprintf("[%d] command: %s", time.Now().Unix(), text))
 		output, err := RunCommand(strings.Split(text, " ")[0], strings.Split(text, " ")[1:])
 		if err != nil {
 			fmt.Printf("[ERROR] cannot run command %s: %v\n", text, err)
+			cmdHistory = append(cmdHistory, fmt.Sprintf("[%d] error: %s", time.Now().Unix(), err.Error()))
+			continue
 		}
 		fmt.Printf(output)
-		cmdHistory = append(cmdHistory, text)
-		// switch text {
-		// case "exit\n":
-		// 	break
-		// case "quit\n":
-		// 	break
-		// default:
-		// 	RunCommand(text, nil)
-		// }
+		cmdHistory = append(cmdHistory, fmt.Sprintf("[%d] output: %s", time.Now().Unix(), output))
 	}
 	WriteFile(".history", cmdHistory)
 	return nil
 }
+
+// HandleTerminationSignal creates a 'listener' on a new goroutine which will notify the
+// program if it receives an interrupt from the OS. We then handle this by calling
+// our clean up procedure and exiting the program.
+func HandleTerminationSignal(cmdHistory *[]string) {
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		fmt.Println("\r- Ctrl+C pressed in Terminal")
+		WriteFile(".history", *cmdHistory)
+		os.Exit(0)
```

Comment:
Only `func main` should be able to terminate the program, so we might do this handling in `main`, and run `Run` asychronously, for example.

Reply:
but what if someone is using my package and calls Run directly the sigterm should be handled, no?

Reply:
In general, I think it's up to the owner of `func main` to handle signals. What you could do is return a `cancel` or `shutdown` function which can be called on such a signal, to save your data.

Reply:
I could not do it. So, I've just avoided the app to crash. Let's go over this on Thursday.